src/README.md

```diff
@@ -0,0 +1,26 @@
+  * In the single-threaded case, we take a simpler approach by pooling any column with `pool` value `0.0 < pool <= 1.0`, meaning even if we're not totally sure the column will be pooled, we'll pool it while parsing, then decide post-parsing whether the column should actually be pooled or unpooled.
```

Comment:
can you comment why a similar approach is not used for multi-threaded parsing? I.e. we could pool per thread and only afterwards - before merging pools - use some heuristics to decide if pools should be merged (this should be more accurate than sampling based decision).

I am not super convinced that we have to change it. In general I would put a recommendation somewhere that users should preferably manually signal if the column should be pooled or not by `Bool` argument.

Reply:
Yes, manual is always better. The reason we determine pooling during sampling is performance; it's a little more expensive to pool a column while parsing, because we have to call `hash` and do `Dict` lookup in addition to setting array value of ref code. So ideally, we only pay that extra cost when we know we will pool.